### ISSUE TITLE
Plans: can't compare at /plans/compare (All Sites) due to Free Trials JS error

### ISSUE THREAD

**Issue by lancewillett** [Automattic/wp-calypso#2347]
Can't view plans comparison in current wpcalypso build.

![screen shot 2016-01-12 at 13 52 42](https://cloud.githubusercontent.com/assets/66797/12276598/e9314864-b933-11e5-8ec1-e83273fe1f15.png)

**Steps to repeat:**

1. Load https://wpcalypso.wordpress.com/plans/compare
2. View the placeholder not replace with data.

I tested with Safari 9.0.2 on OS X 10.11.1.

**JS error:**

```
showFreeTrialExceptionupgrades.3a7a4fd32979cfd4c38d.min.js

TypeError: undefined is not an object (evaluating 'this.props.sites.getSelectedSite().plan.free_trial')
showFreeTrialExceptionupgrades.3a7a4fd32979cfd4c38d.min.js:5:2998
showFreeTrialException
freeTrialExceptionMarkerupgrades.3a7a4fd32979cfd4c38d.min.js:5:3479
freeTrialExceptionMarker
(anonymous function)upgrades.3a7a4fd32979cfd4c38d.min.js:5:2477
map
```

**Comment by lancewillett** [comment]
Note, single site compare seems to work just fine. For example, /plans/compare/lancetest23.wordpress.com

**Comment by lancewillett** [comment]
Flow notes, for how I got to this place:

I got here after renewing a plan.

<img width="877" alt="view-plan-features" src="https://cloud.githubusercontent.com/assets/66797/12276848/4759f818-b935-11e5-82a8-a5e732f03f16.png">

URL: `/purchases/example.wordpress.com/123456` where "123456" is the purchase ID.

There's a link in the page that says: "View Plan Features" and links to `plans/compare` rather than the compare view for just this site.
